lib/open_project/scm/manageable_repository.rb

```diff
@@ -30,32 +30,47 @@
 module OpenProject
   module Scm
     module ManageableRepository
-      ##
-      # We let SCM vendor implementation define their own
-      # types (e.g., for differences in the management of
-      # local vs. remote repositories).
-      #
-      # But if they are manageable by OpenProject, they must
-      # expose this type through +available_types+.
-      MANAGED_TYPE = :managed
-      ##
-      # Reads from configuration whether new repositories of this kind
-      # may be managed from OpenProject.
-      def manageable?
-        !managed_root.nil?
+      def self.included(base)
+        base.extend(ClassMethods)
+      end
+
+      module ClassMethods
+        ##
+        # We let SCM vendor implementation define their own
+        # types (e.g., for differences in the management of
+        # local vs. remote repositories).
+        #
+        # But if they are manageable by OpenProject, they must
+        # expose this type through +available_types+.
+        def managed_type
+          :managed
+        end
+
+        ##
+        # Reads from configuration whether new repositories of this kind
+        # may be managed from OpenProject.
+        def manageable?
+          ! (disabled_types.include?(managed_type) || managed_root.nil?)
```

Comment:
I start to see your point. `manageable?` should from now on only matter when creating a repository. The agreement we made concerning managed repositories was that you can *always* use and delete previously created managed repositories, even if the user disabled them in the meantime.

I'll have to review the code based on these changes a bit more to make clear the subtle ramifications of the changes in this code.

I would still suggest we let `manageable?` be the only point where code outside of repositories may decide if:

* The `:manages` configuration has been set
* OR whether the managed repository has been disabled,

as I do not want to have two checks for that specific purpose in all managed-related code (the two services, and the controller).